packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -171,6 +171,19 @@ export class SelectedNetworkController extends BaseController<
         state.domains[METAMASK_DOMAIN] = networkClientId;
       }
     });
+
+    if (!this.state.perDomainNetwork && domain === METAMASK_DOMAIN) {
+      Object.entries(this.state.domains).forEach(
+        ([entryDomain, networkClientIdForDomain]) => {
+          if (entryDomain !== 'metamask') {
```

Comment:
should this not be `entryDomain === 'metamask'` instead? as in we want to skip if the origin is metamask?

Reply:
if so, i'm not sure that this check to skip is necessary as `state.domains[METAMASK_DOMAIN]` would have been set to `networkClientId` already and would not be called again due to the `(networkClientIdForDomain !== networkClientId)` guard below this

Reply:
I switched it up a bit cause it was defs incorrect how it was =\